src/rpp/sockets.cpp

```diff
@@ -2036,18 +2036,18 @@ namespace rpp
 
         try
         {
-            Class Activity{"android/app/Activity"};
-            Class ConnectivityManager{"android/net/ConnectivityManager"};
-            Class LinkProperties{"android/net/LinkProperties"};
-            Class Network{"android/net/Network"};
-
-            Method getSystemService = Activity.Method("getSystemService", "(Ljava/lang/String;)Ljava/lang/Object;");
-            Method getAllNetworks = ConnectivityManager.Method("getAllNetworks", "()[Landroid/net/Network;");
-            Method getLinkProperties = ConnectivityManager.Method("getLinkProperties", "(Landroid/net/Network;)Landroid/net/LinkProperties;");
-            Method getInterfaceName = LinkProperties.Method("getInterfaceName", "()Ljava/lang/String;");
-            Method getNetworkHandle = Network.Method("getNetworkHandle", "()J");
-
-            Ref<jobject> connectivityManager = getSystemService.Object(getActivity(), MakeString("connectivity").get());
+            static Class Activity{"android/app/Activity"};
+            static Class ConnectivityManager{"android/net/ConnectivityManager"};
+            static Class LinkProperties{"android/net/LinkProperties"};
+            static Class Network{"android/net/Network"};
+
+            static Method getSystemService = Activity.Method("getSystemService", "(Ljava/lang/String;)Ljava/lang/Object;");
+            static Method getAllNetworks = ConnectivityManager.Method("getAllNetworks", "()[Landroid/net/Network;");
+            static Method getLinkProperties = ConnectivityManager.Method("getLinkProperties", "(Landroid/net/Network;)Landroid/net/LinkProperties;");
+            static Method getInterfaceName = LinkProperties.Method("getInterfaceName", "()Ljava/lang/String;");
+            static Method getNetworkHandle = Network.Method("getNetworkHandle", "()J");
+
+            static Ref<jobject> connectivityManager = getSystemService.Object(getActivity(), MakeString("connectivity").get());
```

Comment:
This is a bug I think, because it needs to be converted to a global ref - I'm not sure how JVM GC handles local refs that live forever.

Use the `.to_global()` method on the Ref<T>